### PR TITLE
design-system/feature/create-three-dot-loading-indicator

### PR DESCRIPTION
**Pull request by devsadeq**
- Create three dot loading indicator.
- Add loading state to Bp button.
- Handle loading at dashoard and support login screens.

java_i4rX7HsPgI

### CODE REVIEW

**Comment by mosayed01** [comment]
Could you please provide a simple demo of that loading? @devsadeq

**Comment by devsadeq** [comment]
@mosayed01 check it now.

File: design_system/shared/src/commonMain/kotlin/ui/composable/BpThreeDotLoadingIndicator.kt [hunk 1]
```diff
@@ -0,0 +1,64 @@
...
+                    repeatMode = RepeatMode.Reverse
+                ),
+            )
+        }
+    }
+
+    Row {
+        circles.forEach { animatable ->
+            Box(
+                modifier = modifier
```
Comment (mosayed01) [inline design_system/shared/src/commonMain/kotlin/ui/composable/BpThreeDotLoadingIndicator.kt:56]: I think the default modifier should pass to root composable.. in your case should pass to Row composable not Box @devsadeq

**Review by aya-seif — APPROVED** [review]

File: design_system/shared/src/commonMain/kotlin/ui/composable/BpThreeDotLoadingIndicator.kt [hunk 2]
```diff
@@ -0,0 +1,64 @@
...
+    dotPadding: Dp = 2.dp,
+    dotColor: Color = Theme.colors.onPrimary,
+    animationDelay: Int = 600,
+    initialAlpha: Float = 0.38f,
+) {
+
+    val circles = listOf(
+        remember { Animatable(initialValue = initialAlpha) },
+        remember { Animatable(initialValue = initialAlpha) },
+        remember { Animatable(initialValue = initialAlpha) }
```
Comment (mosayed01) [inline design_system/shared/src/commonMain/kotlin/ui/composable/BpThreeDotLoadingIndicator.kt:37]: I guess there is a better way to handle this case using 1 Animatable 
Lets Assume we have one Animatable
Take value from initial alpha to one alpha
So may we can use this single animatble to handle all dots 
Like when first dot was in initial alpha the second dot was in initial alpha+someRatio.
May u could find better way than it. I trust U will ❤️.
Thanks for reading @devsadeq

**Review by hassanwasfy — APPROVED** [review]
hhh i liked the idea, keep going sadeq

**Review by ay0o5h — APPROVED** [review]

**Review by ahmednasserzaza — APPROVED** [review]
Well done 👏

File: design_system/shared/src/commonMain/kotlin/ui/composable/BpThreeDotLoadingIndicator.kt [hunk 3]
```diff
@@ -0,0 +1,64 @@
...
+fun BpThreeDotLoadingIndicator(
+    modifier: Modifier = Modifier,
+    dotSize: Dp = 12.dp,
+    dotPadding: Dp = 2.dp,
+    dotColor: Color = Theme.colors.onPrimary,
+    animationDelay: Int = 600,
+    initialAlpha: Float = 0.38f,
+) {
+
+    val circles = listOf(
```
Comment (mosayed01) [inline design_system/shared/src/commonMain/kotlin/ui/composable/BpThreeDotLoadingIndicator.kt:34]:
```suggestion
    val circles = remeber { mutableStateListOf(
```
I guess it's little better

Comment (devsadeq) [reply design_system/shared/src/commonMain/kotlin/ui/composable/BpThreeDotLoadingIndicator.kt:34]: @mosayed01 I tried to using only one animatable and that make the code more complex, so do you have any suggestions?

Comment (mosayed01) [reply design_system/shared/src/commonMain/kotlin/ui/composable/BpThreeDotLoadingIndicator.kt]: U can make it simple now and we could refactor in the future. Good job please merge the PR 👏 @devsadeq

**Review by mosayed01 — APPROVED** [review]

**Pull request merged**